Neos.Flow/Classes/Security/Policy/Roles.php

```diff
@@ -0,0 +1,98 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Security\Policy;
+
+use Neos\Flow\Annotations as Flow;
+
+/*
+* This file is part of the Neos.FluidAdaptor package.
+*
+* (c) Contributors of the Neos Project - www.neos.io
+*
+* This package is Open Source Software. For the full copyright and license
+* information, please view the LICENSE file which was distributed with this
+* source code.
+*/
+
+/**
+ * A collection class containing all roles for a account
+ *
+ * @Flow\Proxy(false)
+ */
+final class Roles implements \JsonSerializable, \IteratorAggregate, \Countable, \ArrayAccess
+{
+    /**
+     * @var Role[]
+     */
+    private $roles;
+
+    private function __construct(array $roles)
+    {
+        $this->roles = $roles;
+    }
+
+    public static function create(): self
+    {
+        return new static([]);
+    }
+
+    public static function fromArray(array $roles): self
+    {
+        $processedRoles = [];
+        array_walk($roles, function ($role) use (&$processedRoles) {
+            if (!$role instanceof Role) {
+                throw new \InvalidArgumentException('Expected instance of Role. Was given ' . gettype($role));
+            }
+            $processedRoles[(string)$role] = $role;
+        });
+        return new static($processedRoles);
+    }
+
+    public function count(): int
+    {
+        return count($this->roles);
+    }
+
+    public function has(Role $role): bool
```

Comment:
`$account->getRoles()->hasRole(new Role('Some.Role'))` is weird
`$account->getRoles()->has('Some.Role')` would be better IMO

Reply:
There would be `isset($account->getRoles()['Some.Role'])` but that also feels weird. Changed the method to accept both `Role` and string.